packages/nuxt/src/app/components/utils.ts

```diff
@@ -4,6 +4,7 @@ import type { Component, RendererNode } from 'vue'
 import { isString, isPromise, isArray, isObject } from '@vue/shared'
 import destr from 'destr'
 import type { RouteLocationNormalized } from '#vue-router'
+import { START_LOCATION } from 'vue-router'
```

Comment:
we might need to add a stub for this to avoid including `vue-router` dependency in build when pages integration isn't enabled.

Reply:
Good catch! Also via the `#vue-router` import?

Reply:
We currently use `#build/pages.mjs`.

https://github.com/nuxt/nuxt/blob/7c6b6771ab84686078624bcda089677a680c24cd/packages/nuxt/src/pages/module.ts#L91-L94

I'll push a quick fix.